File: API/APIGeo/APIGeo/Model/Principal.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace APIGeo.Model
{
    public class Principal
    {
        public long Id { get; set; }
        public String Texto { get; set; }
        public virtual ICollection<Label> Labels { get; set; }
        public virtual ICollection<Tag> Tags { get; set; }
    }
}

File: Angular/Geo/src/app/principal/principal.ts
import { Label } from '../label/label';
import { Tag } from '../tag/tag';

export class Principal {
  private id:  number;
  get Id() {		return this.id;  }
  set Id(value:  number) {    this.id = value;  }
  private texto:  string;
  get Texto() {		return this.texto;  }
  set Texto(value:  string) {    this.texto = value;  }

  private labels: Array<Label>;
  get Labels() {		return this.labels;  }
  set Labels(value: Array<Label>) {    this.labels = value;  }

  private tags: Array<Tag>;
  get Tags() {		return this.tags;  }
  set Tags(value: Array<Tag>) {    this.tags = value;  }

}
